### PR TITLE
CLOUDP-155226: include versioned API version in requests

### PR DESCRIPTION
**Pull request by wtrocki**
## What

Use env var provided version when running generator job.

## Why

To fully automate generation we need to supply all input arguments from CI/CD job.
The current version matches the proposal in the specification and it is likely to change.

### CODE REVIEW

File: v2/tools/scripts/generate.sh [hunk 1]
```diff
@@ -4,6 +4,8 @@ set -e
 OPENAPI_FOLDER=${OPENAPI_FOLDER:-"./openapi"}
 OPENAPI_FILE="$OPENAPI_FOLDER/atlas-api.yaml"
 TRANSFORMED_FILE="$OPENAPI_FOLDER/atlas-api-transformed.yaml"
+## Version of the versioned API
+CLIENT_VERSION=${CLIENT_VERSION:-"1.230201.0-dev1"}
```
Comment (gssbzn) [inline v2/tools/scripts/generate.sh:8]: let's keep it YYYYMMDD
```suggestion
CLIENT_VERSION=${CLIENT_VERSION:-"1.20230201.0-dev1"}
```

Comment (wtrocki) [reply v2/tools/scripts/generate.sh:8]: Agreed. Updated spec as well.

File: v2/api/atlas_client.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ const (
 	// CloudURL is default base URL for the services.
 	DefaultCloudURL = "https://cloud.mongodb.com/"
 	// Version the version of the current API client. Should be set to the next version planned to be released.
-	Version    = "2.0.0-dev1"
+	Version    = "1.230201.0-dev1"
```
Comment (gssbzn) [inline v2/api/atlas_client.go:15]: would this be autoupdated? good to do as a follow up tho

**Review by gssbzn — APPROVED** [review]
LGTM

File: v2/api/atlas_client.go (hunk 2 above)
Comment (wtrocki) [reply v2/api/atlas_client.go:15]: I'm going to set env var in the generator job and we should see it. Left it intentionally to have the effect in the PR.

Comment (wtrocki) [reply v2/api/atlas_client.go:15]: Actually merging PR with wrong value seems wrong. Will regenerate it now

Comment (wtrocki) [reply v2/api/atlas_client.go:15]:
```suggestion
	Version    = "1.20230201.0-dev1"
```

**Pull request merged**